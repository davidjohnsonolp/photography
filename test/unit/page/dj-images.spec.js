import '../../../public/src/page/dj-images.js';

let imagesPage;

describe('DJImages Images Page', () => {
  beforeEach(() => {
    imagesPage = window.document.createElement('dj-images'); 
    document.body.appendChild(imagesPage);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-images')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: imagesPage.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});
