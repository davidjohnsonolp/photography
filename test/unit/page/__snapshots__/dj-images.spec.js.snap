// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJImages Images Page renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
          display: flex;
          flex-direction: column;
          flex: 1 0 auto;
        }
        main {
          background: white;
          overflow: hidden;
          flex: 1 0 auto;
          padding: 0 5px;
        }
      </style>
      <dj-header>
      </dj-header>

      <main>
      <dj-h2>Images of beautiful Caledonia</dj-h2>
      <section class=\\"images\\">
          <h3>Choose your image</h3>
          <ul>
              <li>
                  <a href=\\"\\">
                      <img src=\\"\\" alt=\\"Sunset on River Clyde\\">
                  </a>

              </li>
              <li>
                  <a href=\\"\\">
                      <img src=\\"\\" alt=\\"Portree, Isle of Skye\\">
                  </a>
              </li>
              <li>
                  <a href=\\"\\">
                      <img src=\\"\\" alt=\\"Balmaha, Loch Lomond\\">
                  </a>
              </li>
              <li>
                  <a href=\\"\\">
                      <img src=\\"\\" alt=\\"Goatfell, Isle of Arran\\">
                  </a>
              </li>
              <li>
                  <a href=\\"\\">
                      <img src=\\"\\" alt=\\"Ben A'an overlooking Loch Katrine, Trossachs\\">
                  </a>
              </li>
          </ul>
      </section>
      </main>

      <dj-footer>
      </dj-footer>
    ",
}
`;
